### PR TITLE
[Bugfix] Fix projector matrices

### PR DESCRIPTION
**Pull request by Roland-djee**
Projector matrices were computed without filling with identities. Tested with CNOT and SWAP as projector compositions.

### CODE REVIEW

File: tests/qadence/test_matrices.py [hunk 1]
```diff
@@ -470,6 +470,36 @@ def test_projector_tensor(projector: AbstractBlock, exp_projector_mat: Tensor) -
     assert torch.allclose(projector_mat, exp_projector_mat, atol=1.0e-4)
```
Comment (jpmoutinho) [inline tests/qadence/test_matrices.py:470]: I guess this `atol` could be as low as `ATOL_64` no?

File: tests/qadence/test_matrices.py [hunk 2]
```diff
@@ -470,6 +470,36 @@ def test_projector_tensor(projector: AbstractBlock, exp_projector_mat: Tensor) -
...
+            cnot,
+            CNOT(0, 1),
+        ),
+        (swap, SWAP(0, 1)),
+    ],
+)
+def test_projector_composition_unitaries(
+    projector: AbstractBlock, exp_projector: AbstractBlock
+) -> None:
+    assert torch.allclose(block_to_tensor(projector), block_to_tensor(exp_projector))
```
Comment (jpmoutinho) [inline tests/qadence/test_matrices.py:500]: I would also specify the `atol` here.

**Review by jpmoutinho — APPROVED** [review]

File: tests/qadence/test_matrices.py (hunk 2 above)
Comment (Roland-djee) [reply tests/qadence/test_matrices.py:500]: Here, it is possible. Done.

File: tests/qadence/test_matrices.py (hunk 1 above)
Comment (Roland-djee) [reply tests/qadence/test_matrices.py:470]: Nope, this is the best I could achieve for this test.

**Pull request merged**